sunpy/util/xml.py

```diff
@@ -124,3 +127,74 @@ def get_node_text(node):
         else:
             raise NotTextNodeError
     return t
+
+def xml_comments_to_dict(xmlstring):
+    """
+    Parses all the XML comments provided in an XML string into a dictionary
+
+    .. warning::
+
+        For elements with the same name, the last value will be taken.
+
+    Parameters
+    ----------
+    xmlstring : `str`
+        A `str` of xml content.
+
+    Returns
+    -------
+    `dict`
+        The comments present in the string xml input as a dictionary.
+    `str`
+        A concatenated string containing all the comments from the 'HISTORY' element
+
+    Examples
+    --------
+
+    .. code-block:: xml
+
+        <outer>
+            <inner comment = "One">one</inner>
+            <inner comment = "Two">two</inner>
+            <HISTORY comment = "three">two</inner>
+            <HISTORY comment = "four">two</inner>
+        </outer>
+
+    gives you the dict and a string:
+
+    .. code-block:: python
+
+        {u'inner': 'Two'}
+        u'three four'
+    """
+    key_comments_dict = {}
+    history = []
+    node_comments_to_dict(parseString(xmlstring), key_comments_dict, history)
+    history = "".join(history).strip()
+    return key_comments_dict, history
```

Comment:
I'm not a huge fan of this returning two things. Why not just have the key be an input to the function and then call this function twice, once to get the key comments and once to get the history?